ingestion-beam/src/main/java/com/mozilla/telemetry/contextualservices/EmitCounters.java

```diff
@@ -0,0 +1,41 @@
+package com.mozilla.telemetry.contextualservices;
+
+import com.mozilla.telemetry.ingestion.core.Constant.Attribute;
+import com.mozilla.telemetry.metrics.PerDocTypeCounter;
+import com.mozilla.telemetry.transforms.PubsubConstraints;
+import java.util.Map;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.TypeDescriptor;
+
+/**
+ * Extract reporting URL from document and filter out unknown URLs.
+ */
+public class EmitCounters
+    extends PTransform<PCollection<PubsubMessage>, PCollection<PubsubMessage>> {
+
+  public static EmitCounters of() {
+    return new EmitCounters();
+  }
+
+  private EmitCounters() {
+  }
+
+  @Override
+  public PCollection<PubsubMessage> expand(PCollection<PubsubMessage> messages) {
+    return messages.apply(
+        MapElements.into(TypeDescriptor.of(PubsubMessage.class)).via((PubsubMessage message) -> {
+          message = PubsubConstraints.ensureNonNull(message);
+          Map<String, String> attributes = message.getAttributeMap();
+
+          PerDocTypeCounter.inc(attributes, "valid_submission");
```

Comment:
This counter will be redundant with `valid_url`, but more obviously named. We'll transition the existing dashboard to use this rather than `valid_url`.

Reply:
Are you planning to do this as part of the referenced ticket? If not, would you kindly file another ticket?

Reply:
Yes, I'll update the plots in scope of this ticket.